File: web-app/health-insights.js
/**
 * Health Insights Module
 * Provides evidence-based health recommendations based on blink patterns
 * 
 * Medical References:
 * 1. Tsubota, K., & Nakamori, K. (1993). "Dry eyes and video display terminals." 
 *    New England Journal of Medicine, 328(8), 584.
 * 2. Patel, S., et al. (2011). "Normal values for spontaneous blink rate." 
 *    Optometry and Vision Science, 88(9), 1036-1040.
 * 3. Bentivoglio, A. R., et al. (1997). "Analysis of blink rate patterns in normal subjects."
 *    Movement Disorders, 12(6), 1028-1034.
 * 4. Rosenfield, M. (2016). "Computer vision syndrome (a.k.a. digital eye strain)."
 *    Optometry in Practice, 17(1), 1-10.
 */

class HealthInsights {
    constructor() {
        // Normal blink rate ranges (per minute) based on medical literature
        this.NORMAL_BLINK_RATE = {
            min: 12,
            max: 20,
            optimal: 15
        };

        // Thresholds for health alerts
        this.VERY_LOW_THRESHOLD = 8;   // <8 BPM - Severe reduction
        this.LOW_THRESHOLD = 12;        // <12 BPM - Reduced blinking
        this.HIGH_THRESHOLD = 20;       // >20 BPM - Increased blinking
        this.VERY_HIGH_THRESHOLD = 30;  // >30 BPM - Excessive blinking

        // Session duration thresholds (minutes)
        this.SHORT_BREAK_INTERVAL = 20;  // 20-20-20 rule
        this.LONG_BREAK_INTERVAL = 60;   // Hourly break
        this.MAX_CONTINUOUS_WORK = 120;  // 2 hours max

        // Tracking
        this.lastAlertTime = {};
        this.alertCooldown = 300000; // 5 minutes between same alerts
    }

    /**
     * Analyze blink patterns and provide health insights
     */
    analyzeBlinkPattern(stats) {
        const insights = {
            status: 'normal',
            level: 'info',
            title: '',
            message: '',
            recommendations: [],
            medicalNote: '',
            severity: 0, // 0-3: info, warning, alert, critical
            icon: '👁️'
        };

        const bpm = parseFloat(stats.blinksPerMinute) || 0;
        const duration = stats.sessionDuration || 0;
        const durationMinutes = Math.floor(duration / 60);

        // Analyze blink rate
        if (bpm < this.VERY_LOW_THRESHOLD && bpm > 0) {
            insights.status = 'critical';
            insights.level = 'danger';
            insights.severity = 3;
            insights.icon = '🚨';
            insights.title = 'Severely Reduced Blink Rate';
            insights.message = `Your blink rate is ${bpm} per minute, which is significantly below the normal range (12-20 BPM). This is commonly associated with intense screen use.`;
            insights.medicalNote = 'Studies show that blink rate can decrease by up to 60% during computer use, leading to dry eye syndrome (Tsubota & Nakamori, 1993).';
            insights.recommendations = [
                '⚠️ Take an immediate break from the screen',
                '💧 Use artificial tears or lubricating eye drops',
                '👀 Practice conscious blinking exercises',
                '🏥 Consider consulting an eye care professional if symptoms persist',
                '💻 Reduce screen brightness and increase text size',
                '🌡️ Check room humidity (aim for 30-50%)'
            ];
        } else if (bpm < this.LOW_THRESHOLD && bpm > 0) {
            insights.status = 'warning';
            insights.level = 'warning';
            insights.severity = 2;
            insights.icon = '⚠️';
            insights.title = 'Reduced Blink Rate Detected';
            insights.message = `Your blink rate is ${bpm} per minute, below the normal range (12-20 BPM). This may indicate digital eye strain.`;
            insights.medicalNote = 'Normal spontaneous blink rate ranges from 12-20 blinks per minute (Patel et al., 2011). Reduced blinking can lead to tear film instability.';
            insights.recommendations = [
                '😌 Take a 20-second break every 20 minutes',
                '💧 Blink consciously and completely',
                '📏 Follow the 20-20-20 rule: Every 20 minutes, look at something 20 feet away for 20 seconds',
                '💻 Position screen 20-26 inches from eyes',
                '🌊 Stay hydrated - drink water regularly'
            ];
        } else if (bpm > this.VERY_HIGH_THRESHOLD) {
            insights.status = 'alert';
            insights.level = 'warning';
            insights.severity = 2;
            insights.icon = '⚡';
            insights.title = 'Excessive Blink Rate Detected';
            insights.message = `Your blink rate is ${bpm} per minute, significantly above normal (12-20 BPM). This may indicate eye irritation or fatigue.`;
            insights.medicalNote = 'Excessive blinking can be caused by dry eyes, eye irritation, allergies, or eye strain (Bentivoglio et al., 1997).';
            insights.recommendations = [
                '🛑 Take an immediate break from the screen',
                '💧 Check if eyes feel dry or irritated',
                '🌡️ Ensure proper lighting (avoid glare)',
                '🧹 Check for environmental irritants (dust, smoke, dry air)',
                '😎 Consider using blue light filtering glasses',
                '🏥 If persistent, consult an eye care professional'
            ];
        } else if (bpm > this.HIGH_THRESHOLD) {
            insights.status = 'elevated';
            insights.level = 'info';
            insights.severity = 1;
            insights.icon = '📊';
            insights.title = 'Elevated Blink Rate';
            insights.message = `Your blink rate is ${bpm} per minute, slightly above normal (12-20 BPM). This could indicate mild eye fatigue.`;
            insights.medicalNote = 'Slightly elevated blink rates may occur during tasks requiring concentration or in response to environmental factors.';
            insights.recommendations = [
                '😌 Take short breaks periodically',
                '💧 Ensure adequate hydration',
                '🌡️ Check room temperature and humidity',
                '💻 Adjust screen position and brightness'
            ];
        } else if (bpm >= this.NORMAL_BLINK_RATE.min && bpm <= this.NORMAL_BLINK_RATE.max) {
            insights.status = 'healthy';
            insights.level = 'success';
            insights.severity = 0;
            insights.icon = '✅';
            insights.title = 'Healthy Blink Rate';
            insights.message = `Your blink rate is ${bpm} per minute, within the normal range (12-20 BPM). Great job maintaining healthy eye habits!`;
            insights.medicalNote = 'You are maintaining a normal spontaneous blink rate, which helps keep your eyes lubricated and comfortable.';
            insights.recommendations = [
                '👍 Keep up the good habits!',
                '😌 Continue taking regular breaks',
                '💧 Maintain hydration',
                '🌟 Your eye health is on track'
            ];
        }

        // Check session duration
        if (durationMinutes >= this.MAX_CONTINUOUS_WORK) {
            insights.severity = Math.max(insights.severity, 3);
            insights.level = 'danger';
            insights.icon = '🛑';
            insights.title = 'Extended Screen Time Alert';
            insights.message = `You've been working for ${durationMinutes} minutes without a break. Extended screen time increases risk of eye strain.`;
            insights.recommendations.unshift(
                '🚨 TAKE A BREAK NOW - You\'ve exceeded recommended continuous screen time',
                '🚶 Stand up, walk around for 5-10 minutes',
                '👀 Give your eyes a complete rest from screens'
            );
        } else if (durationMinutes >= this.LONG_BREAK_INTERVAL) {
            insights.recommendations.push(
                `⏰ You've been working for ${durationMinutes} minutes - consider taking a longer break soon`
            );
        } else if (durationMinutes >= this.SHORT_BREAK_INTERVAL) {
            insights.recommendations.push(
                '⏰ Time for a 20-second break (20-20-20 rule)'
            );
        }

        return insights;
    }

    /**
     * Get general eye health tips
     */
    getGeneralTips() {
        return {
            title: '👁️ Eye Health Best Practices',
            tips: [
                {
                    title: '20-20-20 Rule',
                    description: 'Every 20 minutes, look at something 20 feet away for 20 seconds',
                    reference: 'American Optometric Association recommendation for reducing digital eye strain'
                },
                {
                    title: 'Proper Lighting',
                    description: 'Ensure room lighting is similar to screen brightness. Avoid glare and harsh overhead lighting',
                    reference: 'Rosenfield, M. (2016). Computer vision syndrome'
                },
                {
                    title: 'Screen Position',
                    description: 'Position screen 20-26 inches from eyes, slightly below eye level (15-20 degrees)',
                    reference: 'OSHA guidelines for computer workstation ergonomics'
                },
                {
                    title: 'Blink Consciously',
                    description: 'Make a conscious effort to blink completely and regularly, especially during screen use',
                    reference: 'Tsubota & Nakamori (1993). Dry eyes and video display terminals'
                },
                {
                    title: 'Humidity Control',
                    description: 'Maintain room humidity between 30-50% to prevent dry eyes',
                    reference: 'American Academy of Ophthalmology recommendations'
                },
                {
                    title: 'Regular Eye Exams',
                    description: 'Get comprehensive eye exams annually, or as recommended by your eye care professional',
                    reference: 'American Academy of Ophthalmology guidelines'
                },
                {
                    title: 'Hydration',
                    description: 'Drink adequate water throughout the day to maintain tear film quality',
                    reference: 'General health recommendation for eye lubrication'
                },
                {
                    title: 'Screen Breaks',
                    description: 'Take a 5-10 minute break every hour from screen work',
                    reference: 'Occupational health guidelines for VDT workers'
                }
            ]
        };
    }

    /**
     * Get symptoms guide for common eye conditions
     */
    getSymptomsGuide() {
        return {
            title: '🏥 Common Eye Symptoms & When to Seek Help',
            conditions: [
                {
                    name: 'Digital Eye Strain (Computer Vision Syndrome)',
                    symptoms: [
                        'Reduced blink rate (<12 BPM)',
                        'Eye fatigue or tiredness',
                        'Dry or irritated eyes',
                        'Blurred vision',
                        'Headaches',
                        'Neck and shoulder pain'
                    ],
                    selfCare: [
                        'Follow 20-20-20 rule',
                        'Adjust screen position and lighting',
                        'Use artificial tears',
                        'Take regular breaks'
                    ],
                    seekHelp: 'If symptoms persist after implementing ergonomic changes'
                },
                {
                    name: 'Dry Eye Syndrome',
                    symptoms: [
                        'Excessive blinking (>20 BPM)',
                        'Burning or stinging sensation',
                        'Redness',
                        'Sensitivity to light',
                        'Feeling of something in the eye'
                    ],
                    selfCare: [
                        'Use preservative-free artificial tears',
                        'Increase humidity in environment',
                        'Avoid direct air flow (fans, AC)',
                        'Stay hydrated'
                    ],
                    seekHelp: 'If symptoms are severe or don\'t improve with over-the-counter treatments'
                },
                {
                    name: 'Eye Fatigue',
                    symptoms: [
                        'Tired, heavy eyes',
                        'Difficulty focusing',
                        'Increased light sensitivity',
                        'Watery eyes'
                    ],
                    selfCare: [
                        'Get adequate sleep (7-9 hours)',
                        'Take frequent breaks',
                        'Ensure proper lighting',
                        'Consider computer glasses'
                    ],
                    seekHelp: 'If fatigue is persistent or affecting daily activities'
                }
            ],
            emergency: {
                title: '🚨 Seek Immediate Medical Attention If:',
                symptoms: [
                    'Sudden vision loss or significant vision changes',
                    'Severe eye pain',
                    'Eye injury',
                    'Flashes of light or floaters',
                    'Double vision',
                    'Red, painful eye with discharge'
                ]
            }
        };
    }

    /**
     * Check if alert should be shown (respects cooldown)
     */
    shouldShowAlert(alertType) {
        const now = Date.now();
        const lastAlert = this.lastAlertTime[alertType] || 0;
        
        if (now - lastAlert > this.alertCooldown) {
            this.lastAlertTime[alertType] = now;
            return true;
        }
        return false;
    }

    /**
     * Get medical disclaimer
     */
    getDisclaimer() {
        return {
            title: '⚕️ Medical Disclaimer',
            content: `
                <strong>IMPORTANT NOTICE:</strong>
                <br><br>
                This application is designed for educational and informational purposes only. 
                It is NOT a substitute for professional medical advice, diagnosis, or treatment.
                <br><br>
                <strong>Key Points:</strong>
                <ul>
                    <li>The health insights provided are based on general medical literature and research</li>
                    <li>Individual health conditions vary significantly</li>
                    <li>This tool does not diagnose medical conditions</li>
                    <li>Always consult qualified healthcare professionals for medical concerns</li>
                    <li>If you experience persistent eye problems, seek professional eye care</li>
                    <li>In case of emergency eye conditions, seek immediate medical attention</li>
                </ul>
                <br>
                <strong>References:</strong>
                <br>
                This application's health recommendations are based on peer-reviewed medical literature 
                including research from the New England Journal of Medicine, Optometry and Vision Science, 
                Movement Disorders, and guidelines from the American Academy of Ophthalmology and 
                American Optometric Association.
                <br><br>
                <strong>Data Privacy:</strong>
                <br>
                All blink data and health insights are stored locally on your device. 
                No health information is transmitted to external servers.
                <br><br>
                <em>By using this application, you acknowledge that you have read and understood this disclaimer.</em>
            `
        };
    }

    /**
     * Get blink rate interpretation
     */
    getBlinkRateInterpretation(bpm) {
        if (bpm === 0) {
            return {
                category: 'No Data',
                description: 'Start tracking to see your blink rate',
                color: '#6c757d'
            };
        } else if (bpm < this.VERY_LOW_THRESHOLD) {
            return {
                category: 'Severely Low',
                description: 'Significant reduction - immediate action recommended',
                color: '#dc3545'
            };
        } else if (bpm < this.LOW_THRESHOLD) {
            return {
                category: 'Below Normal',
                description: 'Reduced blinking - may indicate eye strain',
                color: '#fd7e14'
            };
        } else if (bpm >= this.NORMAL_BLINK_RATE.min && bpm <= this.NORMAL_BLINK_RATE.max) {
            return {
                category: 'Normal',
                description: 'Healthy blink rate',
                color: '#198754'
            };
        } else if (bpm <= this.HIGH_THRESHOLD) {
            return {
                category: 'Slightly Elevated',
                description: 'Mild increase - monitor for patterns',
                color: '#0dcaf0'
            };
        } else if (bpm <= this.VERY_HIGH_THRESHOLD) {
            return {
                category: 'Elevated',
                description: 'Increased blinking - check for irritation',
                color: '#ffc107'
            };
        } else {
            return {
                category: 'Very High',
                description: 'Excessive blinking - may need attention',
                color: '#dc3545'
            };
        }
    }
}

File: src/health_insights.py
"""
Health Insights Module for Eye Blink Tracker
Provides evidence-based health recommendations based on blink patterns

Medical References:
1. Tsubota, K., & Nakamori, K. (1993). "Dry eyes and video display terminals." 
   New England Journal of Medicine, 328(8), 584.
2. Patel, S., et al. (2011). "Normal values for spontaneous blink rate." 
   Optometry and Vision Science, 88(9), 1036-1040.
3. Bentivoglio, A. R., et al. (1997). "Analysis of blink rate patterns in normal subjects."
   Movement Disorders, 12(6), 1028-1034.
4. Rosenfield, M. (2016). "Computer vision syndrome (a.k.a. digital eye strain)."
   Optometry in Practice, 17(1), 1-10.
"""

import logging
import time
from typing import Dict, List, Optional
from dataclasses import dataclass
from enum import Enum


class AlertLevel(Enum):
    """Alert severity levels"""
    INFO = 0
    WARNING = 1
    ALERT = 2
    CRITICAL = 3


@dataclass
class HealthInsight:
    """Health insight data structure"""
    status: str
    level: AlertLevel
    title: str
    message: str
    recommendations: List[str]
    medical_note: str
    icon: str


class HealthInsightsMonitor:
    """Monitor blink patterns and provide health insights"""
    
    # Normal blink rate ranges (per minute)
    NORMAL_BLINK_RATE = {'min': 12, 'max': 20, 'optimal': 15}
    
    # Thresholds for health alerts
    VERY_LOW_THRESHOLD = 8
    LOW_THRESHOLD = 12
    HIGH_THRESHOLD = 20
    VERY_HIGH_THRESHOLD = 30
    
    # Session duration thresholds (minutes)
    SHORT_BREAK_INTERVAL = 20  # 20-20-20 rule
    LONG_BREAK_INTERVAL = 60   # Hourly break
    MAX_CONTINUOUS_WORK = 120  # 2 hours max
    
    def __init__(self):
        self.logger = logging.getLogger(__name__)
        self.last_alert_time = {}
        self.alert_cooldown = 300  # 5 minutes between same alerts
        
    def analyze_blink_pattern(self, blinks_per_minute: float, 
                             session_duration_seconds: int) -> Optional[HealthInsight]:
        """
        Analyze blink patterns and provide health insights
        
        Args:
            blinks_per_minute: Current blink rate
            session_duration_seconds: Duration of current session
            
        Returns:
            HealthInsight object or None if no alert needed
        """
        duration_minutes = session_duration_seconds / 60
        
        # Analyze blink rate
        if blinks_per_minute < self.VERY_LOW_THRESHOLD and blinks_per_minute > 0:
            return self._create_severely_low_alert(blinks_per_minute, duration_minutes)
        elif blinks_per_minute < self.LOW_THRESHOLD and blinks_per_minute > 0:
            return self._create_low_alert(blinks_per_minute, duration_minutes)
        elif blinks_per_minute > self.VERY_HIGH_THRESHOLD:
            return self._create_very_high_alert(blinks_per_minute, duration_minutes)
        elif blinks_per_minute > self.HIGH_THRESHOLD:
            return self._create_elevated_alert(blinks_per_minute, duration_minutes)
        elif duration_minutes >= self.MAX_CONTINUOUS_WORK:
            return self._create_extended_work_alert(duration_minutes)
        elif duration_minutes >= self.LONG_BREAK_INTERVAL:
            return self._create_break_reminder(duration_minutes)
            
        return None
    
    def _create_severely_low_alert(self, bpm: float, duration: float) -> HealthInsight:
        """Create alert for severely reduced blink rate"""
        return HealthInsight(
            status='critical',
            level=AlertLevel.CRITICAL,
            title='Severely Reduced Blink Rate',
            message=f'Your blink rate is {bpm:.1f} per minute, significantly below normal (12-20 BPM). '
                   'This is commonly associated with intense screen use.',
            recommendations=[
                '⚠️ Take an immediate break from the screen',
                '💧 Use artificial tears or lubricating eye drops',
                '👀 Practice conscious blinking exercises',
                '🏥 Consider consulting an eye care professional if symptoms persist',
                '💻 Reduce screen brightness and increase text size',
                '🌡️ Check room humidity (aim for 30-50%)'
            ],
            medical_note='Studies show that blink rate can decrease by up to 60% during computer use, '
                        'leading to dry eye syndrome (Tsubota & Nakamori, 1993).',
            icon='🚨'
        )
    
    def _create_low_alert(self, bpm: float, duration: float) -> HealthInsight:
        """Create alert for reduced blink rate"""
        return HealthInsight(
            status='warning',
            level=AlertLevel.WARNING,
            title='Reduced Blink Rate Detected',
            message=f'Your blink rate is {bpm:.1f} per minute, below normal (12-20 BPM). '
                   'This may indicate digital eye strain.',
            recommendations=[
                '😌 Take a 20-second break every 20 minutes',
                '💧 Blink consciously and completely',
                '📏 Follow the 20-20-20 rule',
                '💻 Position screen 20-26 inches from eyes',
                '🌊 Stay hydrated - drink water regularly'
            ],
            medical_note='Normal spontaneous blink rate ranges from 12-20 blinks per minute '
                        '(Patel et al., 2011). Reduced blinking can lead to tear film instability.',
            icon='⚠️'
        )
    
    def _create_very_high_alert(self, bpm: float, duration: float) -> HealthInsight:
        """Create alert for excessive blink rate"""
        return HealthInsight(
            status='alert',
            level=AlertLevel.ALERT,
            title='Excessive Blink Rate Detected',
            message=f'Your blink rate is {bpm:.1f} per minute, significantly above normal (12-20 BPM). '
                   'This may indicate eye irritation or fatigue.',
            recommendations=[
                '🛑 Take an immediate break from the screen',
                '💧 Check if eyes feel dry or irritated',
                '🌡️ Ensure proper lighting (avoid glare)',
                '🧹 Check for environmental irritants (dust, smoke, dry air)',
                '😎 Consider using blue light filtering glasses',
                '🏥 If persistent, consult an eye care professional'
            ],
            medical_note='Excessive blinking can be caused by dry eyes, eye irritation, allergies, '
                        'or eye strain (Bentivoglio et al., 1997).',
            icon='⚡'
        )
    
    def _create_elevated_alert(self, bpm: float, duration: float) -> HealthInsight:
        """Create alert for elevated blink rate"""
        return HealthInsight(
            status='elevated',
            level=AlertLevel.INFO,
            title='Elevated Blink Rate',
            message=f'Your blink rate is {bpm:.1f} per minute, slightly above normal (12-20 BPM). '
                   'This could indicate mild eye fatigue.',
            recommendations=[
                '😌 Take short breaks periodically',
                '💧 Ensure adequate hydration',
                '🌡️ Check room temperature and humidity',
                '💻 Adjust screen position and brightness'
            ],
            medical_note='Slightly elevated blink rates may occur during tasks requiring concentration '
                        'or in response to environmental factors.',
            icon='📊'
        )
    
    def _create_extended_work_alert(self, duration: float) -> HealthInsight:
        """Create alert for extended screen time"""
        return HealthInsight(
            status='extended_work',
            level=AlertLevel.CRITICAL,
            title='Extended Screen Time Alert',
            message=f'You\'ve been working for {duration:.0f} minutes without a break. '
                   'Extended screen time increases risk of eye strain.',
            recommendations=[
                '🚨 TAKE A BREAK NOW - You\'ve exceeded recommended continuous screen time',
                '🚶 Stand up, walk around for 5-10 minutes',
                '👀 Give your eyes a complete rest from screens',
                '💧 Hydrate and rest your eyes'
            ],
            medical_note='Prolonged screen time without breaks significantly increases the risk of '
                        'computer vision syndrome (Rosenfield, 2016).',
            icon='🛑'
        )
    
    def _create_break_reminder(self, duration: float) -> HealthInsight:
        """Create reminder for taking a break"""
        return HealthInsight(
            status='break_reminder',
            level=AlertLevel.INFO,
            title='Break Reminder',
            message=f'You\'ve been working for {duration:.0f} minutes. Time for a break!',
            recommendations=[
                '⏰ Take a 5-10 minute break',
                '🚶 Stand up and stretch',
                '👀 Look at distant objects',
                '💧 Drink some water'
            ],
            medical_note='Regular breaks help prevent eye strain and maintain productivity.',
            icon='⏰'
        )
    
    def should_show_alert(self, alert_type: str) -> bool:
        """Check if alert should be shown (respects cooldown)"""
        now = time.time()
        last_alert = self.last_alert_time.get(alert_type, 0)
        
        if now - last_alert > self.alert_cooldown:
            self.last_alert_time[alert_type] = now
            return True
        return False
    
    def get_blink_rate_interpretation(self, bpm: float) -> Dict[str, str]:
        """Get interpretation of blink rate"""
        if bpm == 0:
            return {
                'category': 'No Data',
                'description': 'Start tracking to see your blink rate',
                'color': 'gray'
            }
        elif bpm < self.VERY_LOW_THRESHOLD:
            return {
                'category': 'Severely Low',
                'description': 'Significant reduction - immediate action recommended',
                'color': 'red'
            }
        elif bpm < self.LOW_THRESHOLD:
            return {
                'category': 'Below Normal',
                'description': 'Reduced blinking - may indicate eye strain',
                'color': 'orange'
            }
        elif bpm >= self.NORMAL_BLINK_RATE['min'] and bpm <= self.NORMAL_BLINK_RATE['max']:
            return {
                'category': 'Normal',
                'description': 'Healthy blink rate',
                'color': 'green'
            }
        elif bpm <= self.HIGH_THRESHOLD:
            return {
                'category': 'Slightly Elevated',
                'description': 'Mild increase - monitor for patterns',
                'color': 'blue'
            }
        elif bpm <= self.VERY_HIGH_THRESHOLD:
            return {
                'category': 'Elevated',
                'description': 'Increased blinking - check for irritation',
                'color': 'yellow'
            }
        else:
            return {
                'category': 'Very High',
                'description': 'Excessive blinking - may need attention',
                'color': 'red'
            }
    
    @staticmethod
    def get_general_tips() -> List[Dict[str, str]]:
        """Get general eye health tips"""
        return [
            {
                'title': '20-20-20 Rule',
                'description': 'Every 20 minutes, look at something 20 feet away for 20 seconds',
                'reference': 'American Optometric Association recommendation'
            },
            {
                'title': 'Proper Lighting',
                'description': 'Ensure room lighting is similar to screen brightness',
                'reference': 'Rosenfield, M. (2016). Computer vision syndrome'
            },
            {
                'title': 'Screen Position',
                'description': 'Position screen 20-26 inches from eyes, slightly below eye level',
                'reference': 'OSHA guidelines for computer workstation ergonomics'
            },
            {
                'title': 'Blink Consciously',
                'description': 'Make a conscious effort to blink completely and regularly',
                'reference': 'Tsubota & Nakamori (1993). Dry eyes and video display terminals'
            },
            {
                'title': 'Humidity Control',
                'description': 'Maintain room humidity between 30-50%',
                'reference': 'American Academy of Ophthalmology recommendations'
            },
            {
                'title': 'Regular Eye Exams',
                'description': 'Get comprehensive eye exams annually',
                'reference': 'American Academy of Ophthalmology guidelines'
            }
        ]
    
    @staticmethod
    def get_medical_disclaimer() -> str:
        """Get medical disclaimer text"""
        return """
MEDICAL DISCLAIMER

This application is designed for educational and informational purposes only. 
It is NOT a substitute for professional medical advice, diagnosis, or treatment.

Key Points:
• The health insights provided are based on general medical literature and research
• Individual health conditions vary significantly
• This tool does not diagnose medical conditions
• Always consult qualified healthcare professionals for medical concerns
• If you experience persistent eye problems, seek professional eye care
• In case of emergency eye conditions, seek immediate medical attention

References:
This application's health recommendations are based on peer-reviewed medical literature 
including research from the New England Journal of Medicine, Optometry and Vision Science, 
Movement Disorders, and guidelines from the American Academy of Ophthalmology and 
American Optometric Association.

Data Privacy:
All blink data and health insights are stored locally on your device. 
No health information is transmitted to external servers.

By using this application, you acknowledge that you have read and understood this disclaimer.
"""
